server/src/main/java/org/apache/druid/server/ClientQuerySegmentWalker.java

```diff
@@ -218,6 +231,7 @@ public <T> QueryRunner<T> getQueryRunnerForIntervals(final Query<T> query, final
       // Dry run didn't go well.
       throw new ISE("Cannot handle subquery structure for dataSource: %s", query.getDataSource());
     }
+    assert toolChest.canExecuteFully(query.withDataSource(inlineDryRun));
```

Comment:
Let's not commit the assert.  I've seen asserts show up in profiling of production systems before.  I know that they are supposed to be noops and I have no idea how to explain why I've seen them show up.  If it's truly important to check this here, then it should be an if with an exception, if production ever fails from this assert, an `AssertionError` doesn't really provide much help.